test/src/test/java/org/corfudb/runtime/checkpoint/MockLivenessUpdater.java

```diff
@@ -0,0 +1,103 @@
+import org.corfudb.runtime.collections.TableOptions;
+import org.corfudb.runtime.collections.TxnContext;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+@Slf4j
+public class MockLivenessUpdater implements LivenessUpdater {
+
+    private ScheduledExecutorService executorService;
+    private static final int updateInterval = 250;
+
+    private Table<TableName, ActiveCPStreamMsg, Message> activeCheckpointsTable = null;
+    private Table<TableName, CheckpointingStatus, Message> checkpointingStatusTable = null;
+
+    private final CorfuStore corfuStore;
+
+    private TableName tableName = null;
+
+    public MockLivenessUpdater(CorfuStore corfuStore) {
+        this.corfuStore = corfuStore;
+        try {
+            this.activeCheckpointsTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                    DistributedCompactor.ACTIVE_CHECKPOINTS_TABLE_NAME,
+                    TableName.class,
+                    ActiveCPStreamMsg.class,
+                    null,
+                    TableOptions.fromProtoSchema(ActiveCPStreamMsg.class));
+
+            this.checkpointingStatusTable = this.corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                    DistributedCompactor.CHECKPOINT_STATUS_TABLE_NAME,
+                    TableName.class,
+                    CheckpointingStatus.class,
+                    null,
+                    TableOptions.fromProtoSchema(CheckpointingStatus.class));
+        } catch (Exception e) {
+            log.error("Caught an exception while opening metadata tables ", e);
+        }
+    }
+
+    @Override
+    public void updateLiveness(TableName tableName) {
+        this.tableName = tableName;
+        // update validity counter every 250ms
+        executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.scheduleWithFixedDelay(() -> {
+            try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                ActiveCPStreamMsg currentStatus =
+                        txn.getRecord(activeCheckpointsTable, tableName).getPayload();
+                ActiveCPStreamMsg newStatus = ActiveCPStreamMsg.newBuilder()
+                        .setSyncHeartbeat(currentStatus.getSyncHeartbeat() + 1)
+                        .build();
+                txn.putRecord(activeCheckpointsTable, tableName, newStatus, null);
+                txn.commit();
+                log.info("Updated liveness for table {} to {}", tableName, currentStatus.getSyncHeartbeat() + 1);
+            } catch (Exception e) {
+                log.error("Unable to update liveness for table: {}, e ", tableName, e);
+            }
+        }, 0, updateInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private void changeStatus() {
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
```

Comment:
I see a lot of code that assumes transactions cannot fail on conflicts which isn't a safe assumption. Most of these transactions should try to retry a couple of times and then fail.

Reply:
this is the mock framework for tests right? Do you mean to add try catch in the main code?